haero/testing.cpp

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2021, National Technology & Engineering Solutions of Sandia,
+// LLC (NTESS). Copyright (c) 2022, Battelle Memorial Institute
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "testing.hpp"
+
+#include <ekat/ekat_session.hpp>
+
+namespace haero {
+
+namespace {
+
+using Real = haero::Real;
+
+// A simple memory allocation pool for standalone ColumnViews to be used in
+// (e.g.) unit tests. A ColumnPool manages a number of ColumnViews with a fixed
+// number of vertical levels.
+class ColumnPool {
+  size_t num_levels_;          // number of vertical levels per column (fixed)
+  size_t num_cols_;            // number of allocated columns
+  std::vector<int> col_used_;  // columns that are being used already
```

Comment:
Why `int` instead of `bool`? Equivalent, but just curious if there's a clever reason.

Reply:
Glad you asked! https://stackoverflow.com/questions/17794569/why-isnt-vectorbool-a-stl-container

Reply:
![ryan-gosling-slow-clapping-q1x52yju6t9wn39v](https://user-images.githubusercontent.com/28991064/231526864-f1814ca2-8385-49a0-b46e-d98daa813d8e.gif)